kalite/coachreports/static/js/coachreports/coach_reports/views.js

```diff
@@ -0,0 +1,343 @@
+            var data = this.model.get("logs")[this.contents.at(i).get("id")];
+            var new_view = this.add_subview(TabularReportRowCellView, {model: new Backbone.Model(data)});
+            cell_views.push(new_view);
+            this.listenTo(new_view, "detail_view", this.show_detail_view);
+        }
+        this.append_views(cell_views);
+    },
+
+    show_detail_view: function(model) {
+        if (this.detail_view) {
+            // TODO (rtibbles): Implement Models properly here to reflect server side id attributes
+            if (this.detail_view.model.cid === model.cid) {
+                delete this.detail_view;
+                this.trigger("detail_view");
+                return false
+            }
+            this.detail_view.remove();
+        }
+
+
+        var model_id = model.get("exercise_id") || model.get("video_id") || model.get("content_id");
+        this.detail_view = new DetailPanelInlineRowView({
+            model: model,
+            contents_length: this.contents.length,
+            content_item: this.contents.find(function(item) {return item.get("id") === model_id;})
+        });
+        this.$el.after(this.detail_view.el);
+
+        this.trigger("detail_view", this.detail_view);
+
+    }
+
+});
+
+var TabularReportRowCellView = BaseView.extend({
+
+    tagName: 'td',
+
+    events: {
+        "click": "show_detail_view"
+    },
+
+    status_class: function() {
+        var status_class = "partial";
+        if (_.isEmpty(this.model.attributes)) {
+            status_class = "not-attempted";
+        } else if (this.model.get("complete")) {
+            status_class = "complete";
+        } else if (this.model.get("struggling")) {
+            status_class = "struggling";
+        }
+        return status_class;
+    },
+
+    className: function() {
+        return sprintf("status data %s", this.status_class());
+    },
+
+    title_attributes: {
+        "not-attempted": gettext("Not Attempted"),
+        "partial": gettext("Attempted"),
+        "complete": gettext("Complete"),
+        "struggling": gettext("Struggling")
+    },
+
+    attributes: function() {
+        return {title: this.title_attributes[this.status_class()]};
+    },
+
+    initialize: function() {
+        this.render();
+    },
+
+    render: function() {
+        if (this.model.has("streak_progress")) {
+            this.$el.html(this.model.get("streak_progress") + "%");
+        }
+    },
+
+    show_detail_view: function() {
+        if (_.isEmpty(this.model.attributes)) {
+            return false;
+        } else {
+            this.trigger("detail_view", this.model);
+        }
+    }
+});
+
+var FacilitySelectView = Backbone.View.extend({
+    template: HB.template('coach_nav/facility-select'),
+
+    initialize: function() {
+        _.bindAll(this);
+        this.facility_list = new FacilityCollection();
+        this.facility_list.fetch();
+        this.listenTo(this.facility_list, 'sync', this.render);
```

Comment:
What happens in the case of an `error` event (the other possibility from `.fetch()`)? Is there a sensible default in the template, since this element won't be rendered in that case?